internal/controller/progressive/progressive.go

```diff
@@ -380,9 +380,12 @@ func processUpgradingChild(
 			return false, 0, err
 		}
 
-		// if so, create a new upgrading one and mark the existing one for garbage collection
+		///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+		// Replace existing Upgrading child with new one and mark the existing one for garbage collection
+		///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 		if needsUpdating {
-			_, needRequeue, err := startUpgradeProcess(ctx, rolloutObject, existingPromotedChildDef, controller, c)
+			needRequeue := false
+			newUpgradingChildDef, needRequeue, err = startUpgradeProcess(ctx, rolloutObject, existingPromotedChildDef, controller, c)
```

Comment:
set `newUpgradingChildDef` to the new one we just created so we can use it below